packages/flutter/lib/src/material/search.dart

```diff
@@ -108,6 +108,13 @@ abstract class SearchDelegate<T> {
   /// The current value of [query] can be used to determine what the user
   /// searched for.
   ///
+  /// Note that this method might be called multiple times just like any other
+  /// build function in Flutter. Please cache the search results if you don't
+  /// want to call search backend multiple times on the same query.
```

Comment:
A computationally expensive [buildResults] method might not contact a "search backend" (network service). We can only suggest caching here, since exactly what to cache and when will depend on the app. How about this:

This method might be applied more than once to the same query. If your [buildResults] method is computationally expensive, you may want to cache the search results for one or more queries.

Reply:
Done.